### PR TITLE
Feature: enable all overlays

### LINKED ISSUE #27: Add ability to toggle on/off *all* of the overlays at once
When demoing to potential consulting clients, or just trying to explain which applications are active to devs new to single-spa, I like seeing all the overlays at once.

Having a button to toggle them all off/on at once would be cool.

### PR DESCRIPTION
**Pull request by filoxo**
This adds a feature to enable all overlays. Resolves #27 

I struggled with whether I should introduce all of these changes at the same time, this was originally written a pretty long while ago and so includes some other changes. I just haven't had time to sift through them and rebase.

The ToggleGroup might be a bit overengineered, I copied it over from another project.

### CODE REVIEW

**Comment by filoxo** [comment]
single-spa-inspector-showall

**Comment by TheMcMurder** [comment]
I haven't reviewed this yet but I love what I see in the gif! I'll take a look tomorrow unless someone beats me to it.

File: src/inspected-window-helpers/overlay-helpers.js [hunk 1]
```diff
@@ -74,35 +66,49 @@ export function setupOverlayHelpers() {
...
+          ${
+            options.textBlocks && options.textBlocks.length >= 1
+              ? options.textBlocks.map(textBlock => `<div>${textBlock}</div>`)
+              : ""
+          }
+        </div>
+      </div>
+    `;
+    const overlayEl = new DOMParser().parseFromString(domStr, "text/html").body
+      .firstChild;
```
Comment (TheMcMurder) [inline src/inspected-window-helpers/overlay-helpers.js]: I like how easy it is to read. Are there any other benefits to the string that we then parse vs creating objects in memory?

File: src/inspected-window-helpers/overlay-helpers.js [hunk 2]
```diff
@@ -122,14 +128,12 @@ export function setupOverlayHelpers() {
...
+    const [r, g, b] = [
       `0x${hex.slice(0, 2)}`,
       `0x${hex.slice(2, 4)}`,
       `0x${hex.slice(4, 6)}`
-    ];
-    return `rgba(${parseInt(rgba[0])}, ${parseInt(rgba[1])}, ${parseInt(
-      rgba[2]
-    )}, ${opacity})`;
+    ].map(v => parseInt(v));
+    return `rgba(${r}, ${g}, ${b}, ${opacity})`;
```
Comment (TheMcMurder) [inline src/inspected-window-helpers/overlay-helpers.js]: 👍

File: src/panel.html [hunk 3]
```diff
@@ -2,11 +2,20 @@
...
+        background-color: #272822;
+        color: #f8f8f2;
+      }
+    </style>
+    <script></script>
   </head>
   <body style="margin: 0; width: 100%; height: 100%;">
-    <div id="app" style="width:100%; height: 100%">
-      Loading the single-spa inspector...
-    </div>
```
Comment (TheMcMurder) [inline src/panel.html]: Don't we lose the loading state here? Why would we want to lose this?

Comment (filoxo) [reply src/panel.html]: I removed that one in favor of the one rendered by the app itself [panel-app.js#L37-L39](https://github.com/CanopyTax/single-spa-inspector/blob/master/src/panel-app.js#L37-L39). The two messages are different and the other one seems more valuable.

File: src/inspected-window-helpers/overlay-helpers.js (hunk 1 above)
Comment (TheMcMurder) [reply src/inspected-window-helpers/overlay-helpers.js]: I don't know if we need to be too worried about perf. I was just curious if there were other benefits besides the awesome legibility.

File: src/panel.html (hunk 3 above)
Comment (TheMcMurder) [reply src/panel.html]: :heart:

**Comment by joeldenning** [comment]
Ooo! I'm excited for this

**Pull request merged**